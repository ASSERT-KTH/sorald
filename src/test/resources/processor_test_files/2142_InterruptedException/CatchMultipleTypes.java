/*
We should only set the interrupted flag if an interrupt actually occurred, so when a catch has
multiple types in the catch clause, we need to do a type check.
 */
import java.util.concurrent.ExecutionException;

public class CatchMultipleTypes {
    public static void method() {
        try {
            if (1 < 2) {
                throw new ExecutionException(new RuntimeException());
            } else {
                throw new InterruptedException();
            }
        } catch (InterruptedException | ExecutionException e) { // Noncompliant
            throw new RuntimeException(e);
        }
    }
}