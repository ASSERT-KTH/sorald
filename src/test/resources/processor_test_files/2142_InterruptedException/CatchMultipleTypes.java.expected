import java.util.concurrent.ExecutionException;

/**
 We should only set the interrupted flag if an interrupt actually occurred, so when a catch has
 multiple types in the catch clause, we need to do a type check.
 */

public class CatchMultipleTypes {
    public static void method() {
        try {
            if (1 < 2) {
                throw new ExecutionException(new RuntimeException());
            } else {
                throw new InterruptedException();
            }
        } catch (InterruptedException | ExecutionException e) {
            if (e instanceof InterruptedException) {
                Thread.currentThread().interrupt();
            }
            throw new RuntimeException(e);
        }
    }
}